src/Bicep.Decompiler/Rewriters/ParentChildResourceNameRewriter.cs

```diff
@@ -91,7 +92,7 @@ protected override SyntaxBase ReplaceResourceDeclarationSyntax(ResourceDeclarati
                 return syntax;
             }
 
-            foreach (var otherResource in semanticModel.AllResources)
+            foreach (var otherResource in semanticModel.AllResources.OfType<DeclaredResourceMetadata>())
```

Comment:
>DeclaredResourceMetadata [](http://example.com/codeflow?start=76&length=24)

Worth a comment about why we're excluding other types?

Reply:
FYI this pattern is going to appear ALL over the place based on how `ResourceMetadata` is used today. Some code wants to enumerate all of the resources that are *known*, and some code only wants to enumerate all of the declarations in the file. I can comment, but that comment is going to appear a lot so I'd rather just put it somewhere centralized like on `AllResources`.

We could also simplify this by adding a property like `DeclaredResources`.

Reply:
I like the idea of adding an extra property that adds the `OfType<T>()` filter.